### PR TITLE
Add smbusers via pillar

### PR DESCRIPTION
**Pull request by neuhalje**
Add a way to add smbusers via pillar. Basically a cherry-pick of work done by @jbeigh

### CODE REVIEW

File: pillar.example [hunk 1]
```diff
@@ -19,3 +19,15 @@ samba_sections:
...
+    comment: "This is a shared directory"
+    browseable: yes
+    writeable: yes
+    valid users: @sharegroup
+    force group: sharegroup
+    create mask: 0660
+    directory mask: 2770
+
+samba_users:
+   user1: user1sambapassword
```
Comment (gravyboat) [inline pillar.example]: Should we nest passwords below users so it looks like:
```
samba_users:
  user1:
    password: blah
```
 I haven't worked with Samba in a while so I can't remember if there would be any other settings you might need for a user.

Comment (neuhalje) [inline pillar.example]: more properties? uuh, good question, I honestly just need two users on my play-server.

Changing  the value into a map seems like a good idea.

```yaml
 samba_users:
  user1:
    password: blah
    other_option: blubb
```

**Comment by neuhalje** [comment]
Ok, this is the finished PR.

@gravyboat, you can merge

File: pillar.example [hunk 2]
```diff
@@ -19,3 +19,17 @@ samba_sections:
...
+    browseable: yes
+    writeable: yes
+    valid users: @sharegroup
+    force group: sharegroup
+    create mask: 0660
+    directory mask: 2770
+
+samba_users:
+   - login: user1 
+     password: user1sambapassword
```
Comment (gravyboat) [inline pillar.example]: Sorry to comment on this one again, are we going to confuse users with the formatting change between the rest of the file and the `samba_users` section? Nowhere else uses formatting like this.

**Comment by neuhalje** [comment]
@gravyboat  you are thinking of this?

```yaml
 samba_users:
   user1:
    password: blah
    other_option: blubb
   user2:
    password: ...
```

**Comment by gravyboat** [comment]
@neuhalje Yeah definitely.

**Comment by neuhalje** [comment]
@gravyboat Updated (and removed the 'as list' commits)

**Comment by gravyboat** [comment]
Awesome! Thanks @neuhalje, sorry for the lack of direction with my initial comment.

**Pull request merged**